### DISCUSSION TITLE
Misunderstanding about updateNotScheduled

### DISCUSSION THREAD

**Discussion by srudin** [renovatebot/renovate#27585] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

We are having the following configuration:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    ":disableDependencyDashboard",
    ":semanticPrefixFixDepsChoreOthers",
    ":combinePatchMinorReleases",
    ":prHourlyLimit2",
    ":prConcurrentLimit20",
    ":preserveSemverRanges",
    "group:recommended",
    "group:allNonMajor",
    "workarounds:all",
    "schedule:earlyMondays"
  ],
  "regexManagers": [
    {
      "fileMatch": [
        "^Jenkinsfile$"
      ],
      "matchStrings": [
        "[']*(?<depName>[a-z]+)[' ]*: \\[version: '(?<currentValue>[0-9.]+)'\\]"
      ],
      "depTypeTemplate": "jenkins",
      "datasourceTemplate": "maven",
      "description": "Jenkins file service version",
      "depNameTemplate": "ch.post.it.paisa{{#if (equals depName 'platform')}}:paisa-{{depName}}{{else}}.{{depName}}:{{depName}}{{/if}}"
    }
  ],
  "packageRules": [
    {
      "groupName": "all Jenkinsfile dependencies",
      "matchFileNames": ["Jenkinsfile"],
      "automerge": true,
      "automergeType": "branch",
      "schedule": ["at any time"]
    },
    {
      "groupName": "all npm dependencies",
      "matchFileNames": ["**/package.json"]
    },
    {
      "groupName": "all pom dependencies",
      "matchFileNames": ["pom.xml"]
    }
  ],
  "bbUseDefaultReviewers": false,
  "reviewers": ["fankhauseand"]
}
```

As you can see we have defined multiple package rules. With npm RenovateBot is creating one PR for minor and one for major upgrades out-of-the-box plus in our case one for pom upgrades. They should run "earlyMondays". We have defined another package for upgrades in our jenkinsfile which we want to auto-merge immediately.

Basically it's all working fine: We got 3 PRs and the jenkinsfile changes auto-merged. The auto-merge happens right away. Unfortunately the 3 PRs also get updated permanently causing a lot of notifications.

What do we need to do to separate the schedule from the jenkinsfile package from the other ones? It seems that after adding the schedule "at any time" to the jenkinsfile package ALL our packages get updated all the time. I have tried to add separate schedule properties to the other packages (see sample below) but that did NOT help. Is there something wrong in this configuration? Or is this a bug? How can we schedule updates for the 3 PRs "earlyMondays" but the auto-merge "at any time"?

```json
  ...
  "packageRules": [
    {
      "groupName": "all Jenkinsfile dependencies",
      "matchFileNames": ["Jenkinsfile"],
      "automerge": true,
      "automergeType": "branch",
      "schedule": ["at any time"]
    },
    {
      "groupName": "all npm dependencies",
      "matchFileNames": ["**/package.json"],
      "schedule": ["before 4am on Monday"]
    },
    {
      "groupName": "all pom dependencies",
      "matchFileNames": ["pom.xml"],
      "schedule": ["before 4am on Monday"]
    }
  ],
  ...
```

If more information is required please let me know, I'll be happy to provide it if I can.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by srudin** [comment]
PS: Our repository is behind a firewall and I am not allowed to publish the code. I could create a sample repo of course, but before doing that I was hoping that people with a better understanding of RenovateBot than me might see the problem already when looking at the posted configuration.

**Reply by rarkins** [reply, marked as answer]
I'm not sure I understand your description, but are you aware that the `schedule` field controls when branches/PRs are _created_ and not when they are _updated_?

So if you mean "The other PRs were updated out of schedule and I didn't expect that" then that's intended and you can set [updateNotScheduled](https://docs.renovatebot.com/configuration-options/#updatenotscheduled) to `false` to disable it. Also if so then the title of this discussion needs updating to reflect the actual problem

**Reply by srudin** [reply]
Ah no, I wasn't aware that it is possible to configure create/update/merge actions separately, thought everything happens based on the same schedule. Ok then I will try setting :noUnscheduledUpdates, thank you!
